src/wasm_fetcher/https.rs

```diff
@@ -79,4 +86,66 @@ impl Fetcher for Https {
 
         Ok(self.destination.clone())
     }
+
+    async fn fetch_http(&self) -> Result<String> {
+        let http = HttpConnector::new();
+        let client = Client::builder().build::<_, hyper::Body>(http);
+
+        let res = client
+            .get(self.wasm_url.clone().into_string().parse()?)
+            .await?;
+        if res.status() != StatusCode::OK {
+            return Err(anyhow!(
+                "Error while downloading remote WASM module from {}, got HTTP status {}",
+                self.wasm_url,
+                res.status()
+            ));
+        }
+
+        let buf = hyper::body::to_bytes(res).await?;
+        let mut file = File::create(self.destination.clone()).await?;
+        file.write_all(&buf).await?;
+
+        Ok(self.destination.clone())
+    }
+}
+
+#[async_trait]
+impl Fetcher for Https {
+    async fn fetch(&self, sources: &Sources) -> Result<String> {
+        // 1. If CA's provided, download with provided CA's
+        // 2. If no CA's provided, download with system CA's
+        //   2.1. If it fails and if unsafe is enabled for that host,
```

Comment:
nitpicking: replace `unsafe` with `insecure`, which is how the `source.yaml` configuration file calls it